evm/src/modules/CallDispatcher.sol

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.17;
+
+import {ICallDispatcher} from "../interfaces/ICallDispatcher.sol";
+
+
+/// @notice This contract is used to dispatch calls to other contracts.
+contract CallDispatcher is ICallDispatcher {
+    /// @dev funtion returns `success = false` if the target is not a contract and reverts if the call to the target contract fails.
+    function dispatch(address target, bytes calldata data) external returns (bytes memory result, bool success) {
+        if (isContract(target)) {
+            return _dispatch(target, data);
+        }
```

Comment:
I think we should consolidate all this stuff into this `dispatch` call. They're not re-used anywhere